deploy/eck-integrations/values.yaml

```diff
@@ -0,0 +1,348 @@
+## @section Elasticsearch connection parameters
+## @param elasticsearch.host [default: ""] Elasticsearch host
+## @param elasticsearch.user [default: ""] Elasticsearch username
+## @param elasticsearch.pass [default: ""] Elasticsearch password
+## @param elasticsearch.apiKey [default: ""] Elasticsearch access key
+## You may specify an access key instead of username and password
+elasticsearch:
+  host: ""
+  user: ""
+  pass: ""
+  apiKey: ""
```

Comment:
A user may be required to provide a custom CA. Also, it's not great to have to configure sensitive data via Helm values. It would be nice to provide an alternative to avoid this.

If the Elasticsearch storing monitoring data is managed by ECK in the same k8s cluster, then ECK knows how to get all info about it, it just needs the namespace/name. Otherwise for any Elasticsearch, we could use a Secret.

The Elastic `Agent` custom resource has an `elasticsearchRefs` field of `ObjectSelector` type that could help as it supports a custom CA and the 2 usecases (ES managed by ECK and Secret to describe any ES), see the [ObjectSelector doc](https://www.elastic.co/guide/en/cloud-on-k8s/2.11/k8s-api-common-k8s-elastic-co-v1.html#k8s-api-github-com-elastic-cloud-on-k8s-v2-pkg-apis-common-v1-objectselector) and by using it, ECK configures the outputs in the `agent.yml` for you.

```yaml
apiVersion: agent.k8s.elastic.co/v1alpha1
kind: Agent
metadata:
  name: eck-agent-deployment
spec:
  version: 8.12.0

  # Elasticsearch managed by ECK
  elasticsearchRefs:
  - name: es-monitoring
    namespace: observability

  # Any Elasticsearch cluster
  # The referenced secret must contain the following:
  #  - `url`: the URL to reach the Elastic resource
  #  - `username`: the username of the user to be authenticated to the Elastic resource
  #  - `password`: the password of the user to be authenticated to the Elastic resource
  #  - `ca.crt`: the CA certificate in PEM format (optional).
  elasticsearchRefs:
  - secretName: es-monitoring-ref
```

Reply:
thank you @thbkrkr this is more than a valid point and I like the approach of `elasticsearchRefs` for the reasons you mention but also because through that it is easier to support multiple outputs. One thing that I think it is not supported as of now and we need it is being able to utilise an `apikey` and not a pair of `(username, password)`

Reply:
here is a [relevant PR](https://github.com/elastic/cloud-on-k8s/pull/7598) that introduces support for `apikey` only for `Agent->Unmanaged Elastic Search` association